packages/app/src/systems/Bridge/components/AccountSwitch/AccountSwitch.tsx

```diff
@@ -0,0 +1,94 @@
+import { cssObj } from '@fuel-ui/css';
+import {
+  Button,
+  Dialog,
+  Stack,
+  Text,
+  Flex,
+  Copyable,
+  Avatar,
+} from '@fuel-ui/react';
+
+import { shortAddress } from '~/systems/Core/utils';
+
+type AccountSwitchProps = {
+  accounts: string[];
+  onSelect?: (val: string) => void;
+  onConnect?: () => void;
+  onDisconnect?: () => void;
+};
+
+export const AccountSwitch = ({
+  accounts,
+  onSelect,
+  onConnect,
+  onDisconnect,
+}: AccountSwitchProps) => {
+  return (
+    <>
+      <Dialog.Close />
+      <Dialog.Heading css={styles.heading}>
+        <Stack>
+          {accounts.map((account) => {
+            return (
+              <Flex
+                align="center"
+                gap="$1"
+                key={account}
+                css={onSelect && styles.hover}
+              >
+                <Avatar.Generated
+                  size={'xsm'}
+                  background="fuel"
+                  hash={account}
+                />
+                <Text
+                  onClick={() => onSelect && onSelect(account)}
```

Comment:
onSelect should be triggered when clicking anywhere in the line (same as hover), and not only in text